nautobot/core/graphql/schema.py

```diff
@@ -228,40 +228,38 @@ def extend_schema_type_custom_field(schema_type, model):
         prefix = f"{settings.GRAPHQL_CUSTOM_FIELD_PREFIX}_"
 
     for field in cfs:
-        # 2.0 TODO: #824 replace field.name with field.slug
-        field_name = f"{prefix}{str_to_var_name(field.name)}"
-        if str_to_var_name(field.name) != field.name:
-            # 2.0 TODO: str_to_var_name is lossy, it may cause different fields to map to the same field_name
+        field_key = f"{prefix}{str_to_var_name(field.key)}"
+        if str_to_var_name(field.key) != field.key:
+            # 2.0 TODO: str_to_var_name is lossy, it may cause different fields to map to the same field_key
             # In 2.0 we should simply omit fields whose names/slugs are invalid in GraphQL, instead of mapping them.
             warnings.warn(
```

Comment:
Still in favor of biting the bullet and making this change for 2.0 - just use `{prefix}{field.key}` and if that isn't a "valid" graphql identifier, throw it out instead of converting it. And again, same idea for computed-field and relationship, though I'm fine if we want to do that as a sequel PR.